src/ray/stats/metric.h

```diff
@@ -338,22 +349,22 @@ class Stats {
       return;
     }
     TagsType combined_tags = StatsConfig::instance().GetGlobalTags();
-    // In case that tag containing non-printable chars we replace them to '?'
-    // It's important here because otherwise, the message will fail to be sent.
-    for (auto &t : tags) {
-      for (auto &c : t.second) {
-        if (!isprint(c)) {
-          c = '?';
-        }
+    for (auto &[tag_key, tag_val] : tags) {
+#ifndef NDEBUG
+      // In debug build, verify tag_val is printable.
+      for (auto &c : tag_val) {
+        RAY_CHECK(isprint(c)) << "Found unprintable character " << static_cast<int>(c)
+                              << " in " << tag_val;
       }
-      combined_tags.emplace_back(TagKeyType::Register(t.first), std::move(t.second));
+#endif  // NDEBUG
+      combined_tags.emplace_back(TagKeyType::Register(tag_key), std::move(tag_val));
```

Comment:
Seems this code is very similar to the above code. Could we dedup them?

Reply:
Extracted the logic to a separate function. 

This logic may not run for all possible input in debug mode, but I'm not sure if verification / applying fixes are useful in production mode. Unprintable char in tag value is usually a sign of misconfiguration or data corruption. Exporting these data seem to have low value. I doubt any dashboard will be built with the `?` character in metric / dimension name. And recording these data can result in high memory usage in opencensus. Dropping these data silently as in opencensus seems like the best approach in production. Later on if needed, we can have verification with error logging for 1 in 1000 CheckPrintableChar() calls.

Reply:
Btw this loop costs 1.5~2% of the benchmark.